cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -139,3 +147,40 @@ func (c *Controller) getSpcResource(key string) (*apis.StoragePoolClaim, error)
 	}
 	return spcGot, nil
 }
+
+func syncSpc(spcGot *apis.StoragePoolClaim) error {
+	if len(spcGot.Spec.Disks.DiskList) > 0 {
+		// TODO : reconciliation for manual storagepool provisioning
+		return fmt.Errorf("no reconciliation for manual provisioning for storagepoolclaim %s", spcGot.Name)
+	}
+	glog.Infof("Syncing storagepoolclaim %s", spcGot.Name)
+	// Get kubernetes clientset
+	// namespaces is not required, hence passed empty.
+	newK8sClient, err := k8s.NewK8sClient("")
+	if err != nil {
+		return err
+	}
+	// Get openebs clientset using a getter method (i.e. GetOECS() ) as
+	// the openebs clientset is not exported.
+	newOecsClient := newK8sClient.GetOECS()
+
+	// Get the current count of provisione pool for the storagepool claim
+	cspList, err := newOecsClient.OpenebsV1alpha1().CStorPools().List(metav1.ListOptions{LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spcGot.Name})
```

Comment:
It may not be right to tightly couple SPC with CStorPools. It will cause problems when we need to extend SPC for other CAS (engine) types. Better to do count with StoragePool CR objects instead of CStorPool CR objects.

Reply:
Even if combined, there will be some field in storagepool then, which will indicate the no-of created pools.

Reply:
Ok

Reply:
Fixed